### PR TITLE
Update TypeScript definitions

### PR DESCRIPTION
**Pull request by orta**
Makes it so you don't have to leave your editor to read the docs ❤️

### CODE REVIEW

**Comment by papb** [comment]
This is a great idea! However I have one question: this duplicates the documentation in `README.md` and `types.d.ts`. Do you have any idea on how to prevent them getting out of sync?

File: types.d.ts [hunk 1]
```diff
@@ -83,66 +85,262 @@ type WriteOptions = {
...
+   * @param options 
+   */
   append(path: string, data: AppendData, options?: AppendOptions): void;
+  
+  /**
+   * Appends given data to the end of file. If file or any parent directory doesn't exist it will be created.
+   * 
+   * @param path  the path to file.  
+   * @param data  data to append (can be `String` or `Buffer`).  
+   * @param options
```
Comment (papb) [inline types.d.ts]: How would I know what `options.mode` does and what it defaults to? The docs in readme says:

> options (optional) Object with possible fields:
> 
> mode if the file doesn't exist yet, will be created with given mode. Value could be number (eg. 0o700) or string (eg. '700').

Comment (orta) [reply types.d.ts]: I left all the options docs blank, those docs can go in the interfaces for each option, happy to let someone else add those 👍

**Comment by papb** [comment]
@orta Makes sense, thanks!

**Comment by szwacz** [comment]
Released as v 2.2.3, I also thinkg the duplication of docs won't be that big of an issue.
Thanks!

**Comment by papb** [comment]
@szwacz You released on npm, but it looks like you forgot to create the github release?

**Comment by szwacz** [comment]
There you have it @papb :)

**Pull request merged**